src/riak_kv_group_keys.erl

```diff
@@ -0,0 +1,547 @@
+    GroupParams0 = GroupParams#group_params{backend_mod = BackendMod},
+    ContentFolderFun = content_folder_fun(Bucket, GroupParams0, DbRef, FoldOpts, FoldFun, Acc),
+    Async = proplists:get_bool(async_fold, Opts),
+    case Async of
+        true ->
+            {async, ContentFolderFun};
+        false ->
+            {ok, ContentFolderFun()}
+    end.
+
+to_group_params(PropList) ->
+    #group_params{
+       prefix=to_binary(value_or_default(prefix, PropList, undefined)),
+       delimiter=to_binary(value_or_default(delimiter, PropList, undefined)),
+       start_after=to_binary(value_or_default(start_after, PropList, undefined)),
+       max_keys=to_integer(value_or_default(max_keys, PropList, 1000)),
+       continuation_token=to_binary(value_or_default(continuation_token, PropList, undefined))
+      }.
+
+value_or_default(Key, PropList, Default)->
+    case proplists:get_value(Key, PropList) of
+        <<"">> ->
+            Default;
+        "" ->
+            Default;
+        undefined ->
+            Default;
+        Value ->
+            Value
+    end.
+
+to_binary(undefined) ->
+    undefined;
+to_binary(Param) when is_list(Param) ->
+    list_to_binary(Param);
+to_binary(Param) when is_binary(Param) ->
+    Param.
+
+to_integer(undefined) ->
+    undefined;
+to_integer(Param) when is_list(Param) ->
+    list_to_integer(Param);
+to_integer(Param) when is_binary(Param) ->
+    binary_to_integer(Param);
+to_integer(Param) when is_integer(Param) ->
+    Param.
+
+get_backend(#group_params{backend_mod = BackendMod}) ->
+    BackendMod.
+
+get_prefix(#group_params{prefix = Prefix}) ->
+    Prefix.
+
+get_delimiter(#group_params{delimiter = Delimiter}) ->
+    Delimiter.
+
+get_start_after(#group_params{start_after = StartAfter}) ->
+    StartAfter.
+
+get_max_keys(#group_params{max_keys = MaxKeys}) ->
+    MaxKeys.
+
+get_continuation_token(#group_params{continuation_token = ContinuationToken}) ->
+    ContinuationToken.
+
+-spec set_prefix(group_params(), Prefix::binary()) -> group_params().
+set_prefix(GroupParams, Prefix) ->
+    GroupParams#group_params{prefix = Prefix}.
+
+-spec set_delimiter(group_params(), Delimiter::binary()) -> group_params().
+set_delimiter(GroupParams, Delimiter) ->
+    GroupParams#group_params{delimiter = Delimiter}.
+
+-spec set_start_after(group_params(), StartAfter::binary()) -> group_params().
+set_start_after(GroupParams, StartAfter) ->
+    GroupParams#group_params{start_after = StartAfter}.
+
+-spec set_max_keys(group_params(), MaxKeys::pos_integer()) -> group_params().
+set_max_keys(GroupParams, MaxKeys) ->
+    GroupParams#group_params{max_keys = MaxKeys}.
+
+-spec set_continuation_token(group_params(), ContinuationToken::binary()) ->
+    group_params().
+set_continuation_token(GroupParams, ContinuationToken) ->
+    GroupParams#group_params{continuation_token = ContinuationToken}.
+
+%% NOTE: This  type spec with `fun(() -> no_return())' as the return type is
```

Comment:
Why is the `no_return()` type necessary? That doesn't quite smell right to me....